packages/eslint-plugin-react-hooks/src/RulesOfHooks.ts

```diff
@@ -742,7 +758,8 @@ function getFunctionName(node) {
 /**
  * Convenience function for peeking the last item in a stack.
  */
-
-function last(array) {
-  return array[array.length - 1];
+function last<T>(array: T[]): T {
```

Comment:
This is super minor but could you use the `Array<T>` syntax instead for arrays? It's a convention we also use for the compiler, as our team finds them easier to scan.